### PR TITLE
refactor: Simplify Vault for Offline Storage

### PR DESCRIPTION
**Pull request by alexs-mparticle**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
 - Simplify Vault Implementation to be used as a simple local storage wrapper with a simplified API.

 ## Testing Plan
 - [x] Was this tested locally? If not, explain why.
 - {explain how this has been tested, and what, if any, additional testing should be done}

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-5146

### CODE REVIEW

File: src/vault.ts [hunk 1]
```diff
@@ -121,21 +74,15 @@ export default class Vault<StorableItem extends Dictionary> {
...
 
-        return itemString ? JSON.parse(itemString) : {};
+        return itemString ? JSON.parse(itemString) : null;
     }
 
-    private removeItems(): void {
-        if (!this.offlineStorageEnabled) {
-            return;
-        }
+    private removeItem(): void {
```
Comment (alexs-mparticle) [inline src/vault.ts:85]: Should be `removeFromLocalStorage`

File: src/vault.ts [hunk 2]
```diff
@@ -7,28 +7,19 @@ interface IVaultOptions {
 }
 
 export default class Vault<StorableItem extends Dictionary> {
```
Comment (rmi22186) [inline src/vault.ts:9]: Does this still need to extend dictionary?  I removed it locally, ran the tests and the tests still pass.

File: src/vault.ts [hunk 3]
```diff
@@ -121,21 +74,15 @@ export default class Vault<StorableItem extends Dictionary> {
...
-            return this.contents;
-        }
+    private getFromLocalStorage(): StorableItem | null {
         // TODO: Handle cases where Local Storage is unavailable
         // https://go.mparticle.com/work/SQDSDKS-5022
         const itemString = window.localStorage.getItem(this._storageKey);
 
-        return itemString ? JSON.parse(itemString) : {};
+        return itemString ? JSON.parse(itemString) : null;
     }
```
Comment (rmi22186) [inline src/vault.ts:83]: I think the following would be more TS-y. 

```
const item: string = window.localStorage.getItem(this._storageKey);

return item ? JSON.parse(item) : null;
```

File: test/src/tests-vault.ts [hunk 4]
```diff
@@ -1,329 +1,168 @@
...
 
-    describe('#storeItems', () => {
-        it('should store an array of batches as a hashmap', () => {
-            const batch1 = {
-                foo: 'bar',
+        it('should store batches in the order they were added', () => {
+            const storageKey = 'test-batch-save-order';
+
+            const batch1: Partial<Batch> = {
+                mpid: 'bar',
```
Comment (rmi22186) [inline test/src/tests-vault.ts:52]: Very minor nit.  There should be some sort of order to each mpid and source_request_id for each batch. I know we aren't necessarily referencing those items.  But I might do something like `source_request_id_1/2/3/4/5`, and `mpid1/2/3/4/5` rather than them being totally random. It creates a bit more order and is easier to read.

**Review by rmi22186 — CHANGES_REQUESTED** [review]
Mostly minor nits.  Looks great overall.

File: test/src/tests-vault.ts (hunk 4 above)
Comment (alexs-mparticle) [reply test/src/tests-vault.ts:52]: I think my intention was to show that the order wasn't being alphabetized, which is an issue with JS objects.

Comment (rmi22186) [reply test/src/tests-vault.ts:52]: Oh - let me rephrase - it has more to do with readability so we would know which mpid and which source request id was part of which batch if we ever needed to test those.  I know we're not testing them specifically in this test, but we should adhere to that practice to make it easier to grok that `mpid2` is with `batch2`, `mpid1` is `batch1`, rather than how it is now with mpid of `bar` with `batch1`, mpid `baz` with `batch2`.

**Review by rmi22186 — APPROVED** [review]
